### PR TITLE
Enhancement service exclusions #391

### LINKED ISSUE #391: ServiceExclusion with Collection LineRefs
At this point there are two RoutingConstraint options:

1. ServiceExclusion; having a reference to a service journey pattern, a from scheduled stop point and a to scheduled stop point.
2. RoutingConstraintZone; having a reference to points, and a line.

If one wants to specify the direction on which the routing constraint applies, only a ServiceExclusion is possible.
If one wants to reference a line, instead of a journey pattern, only a RoutingConstraintZone is possible.

I would like to request to add lines (and possibly groupoflines) to ServiceExclusion.

### PR DESCRIPTION
**Pull request by nick-knowles**
Allow SERVICE EXCLUSION to be limited to a LINE
Allow ROUTINE CONSTRAINT ZONE to be restricted to a  LINE or LINE DIRECTION

Fix ROUTING VALIDITY PARAMETERS to include  SERVICE EXCLUSION, ROUTINE CONSTRAINT ZONE and TRANSFER RESTRICTION

fixes #391

### CODE REVIEW

**Comment by ue71603** [comment]
@nick-knowles 
switched the target branch to next (which it should be).

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 1]
```diff
@@ -53,6 +52,9 @@
 				<Date>
 					<Modified>2020-10-05 New modes - add MODE of OPERATION to LINE.</Modified>
 				</Date>
+				<Date>
+					<Modified>2023-12-14 #391 Fix linesInDirection to also allow Allowed line diretcion ref. REmove commented out code</Modified>
```
Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:56]:
```suggestion
					<Modified>2023-12-14 #391 Fix linesInDirection to also allow Allowed line diretcion ref. Remove commented out code</Modified>
```

**Review by ue71603 — APPROVED** [review]

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 2]
```diff
@@ -952,24 +920,42 @@ Rail transport, Roads and Road transport
...
 		</xsd:annotation>
 		<xsd:complexContent>
 			<xsd:extension base="containmentAggregationStructure">
-				<xsd:sequence>
-					<xsd:element ref="LineInDirectionRef" maxOccurs="unbounded"/>
-				</xsd:sequence>
+				<xsd:choice maxOccurs="unbounded">
+					<xsd:element ref="LineInDirectionRef">
+						<xsd:annotation>
+							<xsd:documentation> Reference to LINEs in a specifiic DIRECTION</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:930]:
```suggestion
							<xsd:documentation>Reference to a LINE in a specifiic DIRECTION</xsd:documentation>
```

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 3]
```diff
@@ -952,24 +920,42 @@ Rail transport, Roads and Road transport
...
+						</xsd:annotation>
+					</xsd:element>
+					<xsd:element ref="AllowedLineDirectionRef"/>
+				</xsd:choice>
+			</xsd:extension>
+		</xsd:complexContent>
+	</xsd:complexType>
+	<xsd:complexType name="linesInDirections_RelStructure">
+		<xsd:annotation>
+			<xsd:documentation>Type for a list of LINEsin a specific DIRECTION</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:940]:
```suggestion
			<xsd:documentation>Type for a list of LINEs in a specific DIRECTION</xsd:documentation>
```

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 4]
```diff
@@ -952,24 +920,42 @@ Rail transport, Roads and Road transport
...
+					<xsd:element ref="Line"/>
+					<xsd:element ref="LineInDirectionRef"/>
+				</xsd:choice>
 			</xsd:extension>
 		</xsd:complexContent>
 	</xsd:complexType>
 	<xsd:element name="LineInDirectionRef" type="LineInDirectionRef_Structure" abstract="false">
 		<xsd:annotation>
-			<xsd:documentation> Refrence to LINE in a specific DIRECTION</xsd:documentation>
+			<xsd:documentation> Refrence to LINEs in a specific DIRECTION</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:953]:
```suggestion
			<xsd:documentation>Reference to LINEs in a specific DIRECTION</xsd:documentation>
```

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 5]
```diff
@@ -952,24 +920,42 @@ Rail transport, Roads and Road transport
...
 		</xsd:annotation>
 	</xsd:element>
 	<xsd:complexType name="LineInDirectionRef_Structure">
 		<xsd:annotation>
-			<xsd:documentation>Type for a Reference to a LINE in a specific DIRECTION</xsd:documentation>
+			<xsd:documentation>Type for a Reference to a LINEs in a specific DIRECTION</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:958]:
```suggestion
			<xsd:documentation>Type for a Reference to a LINE in a specific DIRECTION</xsd:documentation>
```

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 6]
```diff
@@ -952,19 +920,37 @@ Rail transport, Roads and Road transport
...
+						<xsd:annotation>
+							<xsd:documentation>Reference to a LINE in a specifiic DIRECTION</xsd:documentation>
+						</xsd:annotation>
+					</xsd:element>
+					<xsd:element ref="AllowedLineDirectionRef"/>
+				</xsd:choice>
+			</xsd:extension>
+		</xsd:complexContent>
+	</xsd:complexType>
+	<xsd:complexType name="linesInDirections_RelStructure">
```
Comment (Aurige) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:938]: Where is this _linesInDirections_RelStructure_ type used ? I cant find it

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 7]
```diff
@@ -952,19 +920,37 @@ Rail transport, Roads and Road transport
...
 		</xsd:annotation>
 		<xsd:complexContent>
 			<xsd:extension base="containmentAggregationStructure">
-				<xsd:sequence>
-					<xsd:element ref="LineInDirectionRef" maxOccurs="unbounded"/>
-				</xsd:sequence>
+				<xsd:choice maxOccurs="unbounded">
+					<xsd:element ref="LineInDirectionRef">
+						<xsd:annotation>
+							<xsd:documentation>Reference to a LINE in a specifiic DIRECTION</xsd:documentation>
```
Comment (Aurige) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:930]:
```suggestion
							<xsd:documentation>Reference to a LINE in a specific DIRECTION</xsd:documentation>
```

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd (hunk 6 above)
Comment (ue71603) [reply xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:938]: @Aurige remove it? Or use it?

**Comment by nick-knowles** [comment]
UMLL updated ; CEN  TM & NeTEx Spec still to do
NeTEx
image

Transmodel
image

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd (hunk 6 above)
Comment (Aurige) [reply xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:963]: @ue71603 remove if not used

**Comment by ue71603** [comment]
@Aurige @skinkie we have to merge it as is and then I will have to remove "linesInDirections_RelStructure". @nick-knowles  did some strange stuff on his branch and it wants to change (fetch, pull) a lot of things before I can commit to the branch. 
image
Do you think it is safe to do?

File: xsd/netex_part_1/part1_tacticalPlanning/netex_routingConstraint_version.xsd [hunk 8]
```diff
@@ -254,12 +315,12 @@ Rail transport, Roads and Road transport
...
 			<xsd:element name="FromPointRef" type="ScheduledStopPointRefStructure" minOccurs="0">
 				<xsd:annotation>
-					<xsd:documentation>From point of restriction.</xsd:documentation>
+					<xsd:documentation>From POINT of TRANSFER RESTRICTION.</xsd:documentation>
 				</xsd:annotation>
 			</xsd:element>
 			<xsd:element name="ToPointRef" type="ScheduledStopPointRefStructure" minOccurs="0">
 				<xsd:annotation>
-					<xsd:documentation>From point of restriction.</xsd:documentation>
+					<xsd:documentation>TO POINT of of TRANSFER RESTRICTION.</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_tacticalPlanning/netex_routingConstraint_version.xsd:323]:
```suggestion
					<xsd:documentation>TO POINT of TRANSFER RESTRICTION.</xsd:documentation>
```

**Pull request merged**